security/self-assessment.md

```diff
@@ -0,0 +1,363 @@
+1. BMO interacts with Ironic using authenticated API calls to execute the actual
+   provisioning operations
+1. IPAM manages IP address allocation requests securely through IPAddress custom
+   resources
+1. All component interactions happen through the Kubernetes API server,
+   inheriting its authentication and authorization controls
+
+The security boundary between components is maintained through:
+
+* Kubernetes RBAC controls for API access
+* Authentication between components
+* Mutual TLS configuration
+* Validation of all inputs before processing
+* Secure storage of sensitive data like BMC credentials in Kubernetes secrets
+
+### Goals
+
+There are a number of great open source tools for bare metal host provisioning,
+including Ironic. Metal3 aims to build on these technologies to provide a
+Kubernetes native API for managing bare metal hosts via a provisioning stack
+that is also running on Kubernetes. We believe that Kubernetes Native
+Infrastructure, or managing your infrastructure just like your applications, is
+a powerful next step in the evolution of infrastructure management.
+
+The Metal3 project is also building integration with the Kubernetes cluster-api
+project, allowing Metal3 to be used as an infrastructure backend for Machine
+objects from the Cluster API. These components integrate seamlessly to leverage
+the Kubernetes ecosystem and automate the provisioning and management of
+bare-metal infrastructure.
+
+### Non-goals
+
+Metal3's non-goals include:
+
+* Non-baremetal hardware provisioning
+* Direct hardware management without Ironic
+* Operating system configuration management after provisioning
+* Network provisioning or SDN implementation
+* Providing a standalone solution outside of Kubernetes
+* Exposing full set of Ironic features
+
+## Self-assessment use
+
+This self-assessment is created by the Metal3 team to perform an internal
+analysis of the project's security.  It is not intended to provide a security
+audit of Metal3, or function as an independent assessment or attestation of
+Metal3's security health.
+
+This document serves to provide Metal3 users with an initial understanding of
+Metal3's security, where to find existing security documentation, Metal3
+plans for security, and general overview of Metal3 security practices, both
+for development of Metal3 as well as security of Metal3.
+
+This document provides the CNCF TAG-Security with an initial understanding of
+Metal3 to assist in a joint-assessment, necessary for projects under
+incubation.  Taken together, this document and the joint-assessment serve as a
+cornerstone for if and when Metal3 seeks graduation and is preparing for a
+security audit.
+
+## Security functions and features
+
+**Critical**
+
+* Ironic: Ironic is the ultimate component which provisions the bare metal
+   hosts, talks to the real hardware and is able to perform very destructive or
+   malicious actions.
+
+**Security Relevant**
```

Comment:
I think we could mention CAPM3 here also. It currently has cluster-wide access to all bare metal hosts as part of its RBAC. (There are initiatives for improving this...)

Reply:
Which all elements contain sensitive information in BMH? UserData? BMC secret is only referenced, right?

Reply:
I was mainly thinking that since it can interact with the BMHs then it can also make the hardware do what it wants. Similar to how Ironic is listed as critical. CAPM3 may not have quite that deep access but it can make the servers boot malicious images for sure.

Reply:
OK, let's add CAPM3.

Reply:
Done in next push.